packages/domains-table/src/domains-table/domains-table-ssl-cell.tsx

```diff
@@ -4,59 +4,64 @@ import { lock } from '@wordpress/icons';
 import clsx from 'clsx';
 import { useTranslate } from 'i18n-calypso';
 
-interface DomainsTableSSLCellProps {
+interface DomainsTableSslCellProps {
 	domainManagementLink: string;
 	sslStatus: DomainData[ 'ssl_status' ];
+	hasWpcomManagedSslCert?: DomainData[ 'has_wpcom_managed_ssl_cert' ];
 }
 
-export default function DomainsTableSSLCell( {
+export default function DomainsTableSslCell( {
 	domainManagementLink,
 	sslStatus,
-}: DomainsTableSSLCellProps ) {
+	hasWpcomManagedSslCert,
+}: DomainsTableSslCellProps ) {
 	const translate = useTranslate();
+	// WordPress.com managed subdomains (e.g. *.wordpress.com, *.wpcomstaging.com, etc.)
+	// are covered by a wildcard SSL cert, so we display them as 'Active'.
+	const isActiveSsl = sslStatus === 'active' || hasWpcomManagedSslCert;
+	const isPendingSsl = sslStatus === 'pending' || sslStatus === 'newly_registered';
+	const domainHasSsl = sslStatus !== null || hasWpcomManagedSslCert;
 
-	const isPendingSSL = sslStatus === 'pending' || sslStatus === 'newly_registered';
-
-	const getSSLStatusText = () => {
-		if ( sslStatus === 'active' ) {
+	const getSslStatusText = () => {
+		if ( isActiveSsl ) {
 			return translate( 'Active' );
 		}
-		if ( isPendingSSL ) {
+		if ( isPendingSsl ) {
 			return translate( 'Pending' );
 		}
 		if ( sslStatus === 'disabled' ) {
 			return translate( 'Disabled' );
 		}
 	};
 
+	let button: React.ReactElement | string;
+
+	if ( sslStatus ) {
+		button = (
+			<a
+				className="domains-table-row__ssl-status-button"
+				href={ `${ domainManagementLink }?ssl-open=true` }
+				onClick={ ( event ) => event.stopPropagation() }
+			>
+				{ getSslStatusText() }
```

Comment:
Optional nit, but since we aren't reusing `getSslStatusText` now, we could potentially just inline the function logic in this if clause. Up to you!

Reply:
Nice catch. I started to make this change but I actually think I prefer to keep the function as it is. The early returns ensuring only one value is rendered feels a little more resilient.

I'm trying to envisions some future change where (perhaps accidentally, due to a regression elsewhere) more than one of those three conditions could be true at the same time.

It feels unlikely, but if it ever did happen somehow, we'd wind up with something like